File: navground_sim/src/info.cpp
/**
 * @author Jerome Guzzi - <jerome@idsia.ch>
 */

#include "navground/core/info.h"
#include "navground/core/behavior.h"
#include "navground/core/behavior_modulation.h"
#include "navground/core/kinematics.h"
#include "navground/sim/build_info.h"
#include "navground/sim/scenario.h"
#include "navground/sim/state_estimation.h"
#include "navground/sim/task.h"
#include "navground/sim/version.h"

int main(int argc, char *argv[]) {
  navground::core::InfoCommand cmd(
      "info", navground::sim::build_info().get_version_string(),
      {{"Behaviors", navground::core::Behavior::type_properties},
       {"Kinematics", navground::core::Kinematics::type_properties},
       {"Modulations", navground::core::BehaviorModulation::type_properties},
       {"State Estimations", navground::sim::StateEstimation::type_properties},
       {"Tasks", navground::sim::Task::type_properties},
       {"Scenarios", navground::sim::Scenario::type_properties}},
      navground::sim::get_build_info(),
      navground::sim::get_build_dependencies());
  return cmd.run(argc, argv);
}

File: navground_sim_py/navground/sim/info.py
import argparse

from navground import sim
from navground.core import command
from navground.core.info import description  # noqa: F401
from navground.core.info import info, init_parser_with_registers
from navground.core.info import registers as core_registers

registers = core_registers + [(sim.StateEstimation, "State Estimations"),
                              (sim.Task, "Tasks"), (sim.Scenario, "Scenarios")]


def init_parser(parser: argparse.ArgumentParser) -> None:
    init_parser_with_registers(parser, registers)


def parser() -> argparse.ArgumentParser:
    parser = argparse.ArgumentParser()
    init_parser(parser)
    return parser


def _main(arg: argparse.Namespace) -> None:
    command._main(arg, sim.load_plugins)
    info(arg, registers, sim.get_build_info(), sim.get_build_dependencies())


def main() -> None:
    arg = parser().parse_args()
    _main(arg)
